test/extensions/filters/http/composite/composite_filter_integration_test.cc

```diff
@@ -153,6 +199,29 @@ TEST_P(CompositeFilterIntegrationTest, TestBasic) {
   }
 }
 
+// Verifies that if we don't match the match action the request is proxied as normal, while if the
+// match action is hit we apply the specified dynamic filter to the stream.
+TEST_P(CompositeFilterIntegrationTest, TestBasicDynamicFilter) {
+  prependCompositeDynamicFilter("composite-dynamic");
+  initialize();
```

Comment:
IIUC the previous discussion, after the `initialize()` the ECDS should also be ready.
Can you please add a wait here that asserts that the dynamic SetResponseCode is ready for use?